dialogue-core/src/main/java/com/palantir/dialogue/core/BalancedScoreTracker.java

```diff
@@ -250,7 +251,9 @@ private static void registerGauges(
                     c -> c.computeScoreSnapshot().getScore(),
                     longStream -> {
                         long[] longs = longStream.toArray();
```

Comment:
stream -> array -> stream

Reply:
I know, it's a big funky, but you can't consume them twice!

Reply:
I wonder if we should filter out zero from channels created and never used. If we have an empty stream, we can be confident the value is zero.

Reply:
(just musing, happy merging as is)